libsolidity/interface/CompilerStack.cpp

```diff
@@ -1248,6 +1248,14 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	if (!compiledContract.yulIR.empty())
 		return;
 
+	if (!_contract.sourceUnit().annotation().useABICoderV2.set() || !*_contract.sourceUnit().annotation().useABICoderV2)
```

Comment:
Does this needs to descend down to all imports?

Reply:
I think in order to be precise yes, otherwise it could be misleading

Reply:
The type checker should ensure that you can just call `!*_contract.sourceUnit().annotation().useABICoderV2` here.

Reply:
Added an assert, but I think we should still go through all the imports.

Any nice way to do that?

Reply:
@axic I think @chriseth meant that the TypeChecker makes sure that this property is transitively applied.

Reply:
I hope that is the case 😅